straxen/common.py

```diff
@@ -76,134 +76,160 @@ def pmt_positions(xenon1t=False):
+    :return: opened file
+    """
+    if file_name in _resource_cache:
         # Retrieve from in-memory cache
-        return _resource_cache[x]
+        return _resource_cache[file_name]
     # File resource
     if fmt in ['npy', 'npy_pickle']:
-        result = np.load(x, allow_pickle=fmt == 'npy_pickle')
+        result = np.load(file_name, allow_pickle=fmt == 'npy_pickle')
         if isinstance(result, np.lib.npyio.NpzFile):
             # Slurp the arrays in the file, so the result can be copied,
             # then close the file so its descriptors does not leak.
             result_slurped = {k: v[:] for k, v in result.items()}
             result.close()
             result = result_slurped
     elif fmt == 'pkl':
-        with open(x, 'rb') as f:
+        with open(file_name, 'rb') as f:
             result = pickle.load(f)
     elif fmt == 'pkl.gz':
-        with gzip.open(x, 'rb') as f:
+        with gzip.open(file_name, 'rb') as f:
             result = pickle.load(f)
     elif fmt == 'json.gz':
-        with gzip.open(x, 'rb') as f:
+        with gzip.open(file_name, 'rb') as f:
             result = json.load(f)
     elif fmt == 'json':
-        with open(x, mode='r') as f:
+        with open(file_name, mode='r') as f:
             result = json.load(f)
     elif fmt == 'binary':
-        with open(x, mode='rb') as f:
+        with open(file_name, mode='rb') as f:
             result = f.read()
     elif fmt == 'text':
-        with open(x, mode='r') as f:
+        with open(file_name, mode='r') as f:
             result = f.read()
     elif fmt == 'csv':
-        result = pd.read_csv(x)
+        result = pd.read_csv(file_name)
     else:
         raise ValueError(f"Unsupported format {fmt}!")
 
     # Store in in-memory cache
-    _resource_cache[x] = result
+    _resource_cache[file_name] = result
 
     return result
 
 
 @export
-def get_resource(x, fmt='text'):
+def get_resource(x: str, fmt='text'):
     """
-    Get the resource from an online source to be opened here.
-    :param x:
-    :param fmt:
-    :return:
+    Get the resource from an online source to be opened here. We will
+        sequentially try the following:
+            1. Load if from memory if we asked for it before;
+            2. load it from a file if the path exists;
+            3. (preferred option) Load it from our database
+            4. Load the file from some URL (e.g. raw github content)
+
+    :param x: str, either it is :
+        A.) a path to the file;
+        B.) the identifier of the file as it's stored under in the database;
+        C.) A URL to the file (e.g. raw github content).
+
+    :param fmt: str, format of the resource x
+    :return: the opened resource file x opened according to the
+        specified format
     """
-    if straxen.uconfig is not None:
+    # 1. load from memory
+    if x in _resource_cache:
+        return _resource_cache[x]
+    # 2. load from file
+    elif os.path.exists(x):
```

Comment:
The `elif` construction doesn't really do anything here, but that's borderline stylistic.

Reply:
valid